test/Test/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Test.Database.LSMTree.Internal.RawPage (
+    -- * Main test tree
+    tests,
+) where
+
+import qualified Data.ByteString.Short as SBS
+import           Data.Primitive.ByteArray (ByteArray (..), byteArrayFromList)
+import qualified Data.Vector.Primitive as V
+import           Data.Word (Word16)
+import           GHC.Word (byteSwap16)
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.HUnit (testCase, (@=?))
+
+import           Database.LSMTree.Internal.RawPage
+import           FormatPage (Key (..), Operation (..), PageLogical (..),
+                     Value (..), encodePage, serialisePage)
+
+tests :: TestTree
+tests = testGroup "Database.LSMTree.Internal.RawPage"
+    [ testCase "empty" $ do
```

Comment:
If empty page is non-compliant, than the doc should be made to explicitly mention that, and also `prototypes` to use `NonEmpty` instead of a list.

Reply:
Ah, I concluded that it wasn't compliant because the [page directory section of `format-page.md`](https://github.com/input-output-hk/lsm-tree/blob/88556326a21ca9362f26b42f0b074f60e3fe1f47/doc/format-page.md#page-directory) implies that `N >= 1` in the line

```md
6. 2*(N+1) when N>1 6 when N=1
```

We should probably change that line then